lib/SPIRV/SPIRVReader.cpp

```diff
@@ -355,11 +356,19 @@ std::string SPIRVToLLVM::transVCTypeName(SPIRVTypeBufferSurfaceINTEL *PST) {
   return VectorComputeUtil::getVCBufferSurfaceName();
 }
 
-Type *SPIRVToLLVM::transType(SPIRVType *T, bool IsClassMember) {
+Type *SPIRVToLLVM::transType(SPIRVType *T, bool UseTPT) {
   auto Loc = TypeMap.find(T);
-  if (Loc != TypeMap.end())
+  if (Loc != TypeMap.end() && !UseTPT)
```

Comment:
I'm a bit confused here: if `UseTPT == true`, does it mean that even if `TypeMap` contains the requested `T`, we will still ignore it and generate a new opaque type?

Won't it create types duplicates due to names collision (like `struct.name`, `struct.name.1`, etc.)?

Reply:
(see below for answer)